src/cmap/handshake/client_metadata.ts

```diff
@@ -234,3 +234,39 @@ export function getFAASEnv(): Map<string, string | Int32> | null {
 
   return null;
 }
+
+/**
+ * @internal
+ * This type represents the global Deno object and the minimal type contract we expect it to satisfy.
+ */
+declare const Deno: { version?: { deno?: string } } | undefined;
+
+/**
+ * @internal
+ * This type represents the global Bun object and the minimal type contract we expect it to satisfy.
+ */
+declare const Bun: { (): void; version?: string } | undefined;
+
+/**
+ * @internal
+ * Get current JavaScript runtime platform
+ *
+ * NOTE: The version information fetching is intentionally written defensively
+ * to avoid having a released driver version that becomes incompatible
+ * with a future change to these global objects.
+ */
+function getRuntimeInfo(): string {
+  if ('Deno' in globalThis) {
+    const version = typeof Deno?.version?.deno === 'string' ? Deno?.version?.deno : '0.0.0-unknown';
```

Comment:
`typeof new String(‘’)` returns “object” so a safer check: https://github.com/mongodb/node-mongodb-native/pull/3637/files#diff-39b2554fd18da165b59a6351b1aafff3714e2a80c1435f2de9706355b4d32351R69

Reply:
Is this something we want to support? Generally, none of our code supports string objects / number objects / etc. I would be suspicious of code that uses String instances.

Unrelated but here is how String objects serialize in BSON, just to exemplify the degree to which we assume these constructors are assumed to not be used:
```js
> BSON.deserialize(BSON.serialize({ a: new String('abc') }))
{ a: { '0': 'a', '1': 'b', '2': 'c' } }
```

Reply:
Didn't know if we wanted to or not, given the serialisation example it seems not then, although unexpected. But I guess if no one has ever reported that no one really uses them.